### PR TITLE
MAGECLOUD-3967: Add static and unit tests to magento-cloud-components

### PR DESCRIPTION
**Pull request by oshmyheliuk**
https://magento2.atlassian.net/browse/MAGECLOUD-3967

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+env:
+  matrix:
+    - TEST_SUITE=static-unit XDEBUG=true
+
+install:
+  - if [ $TRAVIS_SECURE_ENV_VARS == "true" ]; then composer config http-basic.repo.magento.com ${REPO_USERNAME} ${REPO_PASSWORD}; fi;
+  - if [ $TRAVIS_SECURE_ENV_VARS == "true" ]; then composer config github-oauth.github.com ${GITHUB_TOKEN}; fi;
+  - composer config repositories.magento composer https://repo.magento.com/
+  - composer require "magento/framework:*" --no-update
```
Comment (shiftedreality) [inline .travis.yml:19]: This will fail without keys, don't you want to add if statement?

File: composer.json [hunk 2]
```diff
@@ -12,6 +12,27 @@
...
+  },
+  "scripts": {
+    "test": [
+      "@phpstan",
+      "@phpcs",
+      "@phpmd",
+      "@phpunit"
+    ],
+    "phpstan": "phpstan analyse -c Test/static/phpstan.neon",
+    "phpcs": "phpcs ./ --standard=Test/static/phpcs-ruleset.xml -p -n",
```
Comment (shiftedreality) [inline composer.json:29]: Let's add Sniffs from ECE-Tools

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+language: php
+php:
+  - '7.1'
+  - '7.2'
+  - '7.3'
+
+env:
+  matrix:
+    - TEST_SUITE=static-unit XDEBUG=true
```
Comment (shiftedreality) [inline .travis.yml]: Yo don't need matrix, having only 1 variation

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (oshmyheliuk) [reply .travis.yml:19]: It's ok if it will fail, without "magento/framework:*" package test scripts won't run anyway.

File: Model/UrlFixer.php [hunk 4]
```diff
@@ -21,8 +21,7 @@ class UrlFixer
      */
     public function run(Store $store, $url): string
     {
-        if (
-            ($store->getForceDisableRewrites() || !$store->getConfig(Store::XML_PATH_USE_REWRITES))
+        if (($store->getForceDisableRewrites() || !$store->getConfig(Store::XML_PATH_USE_REWRITES))
             && strpos($url, '/magento/') !== false
```
Comment (NadiyaS) [inline Model/UrlFixer.php:27]: I would also create task to core to fix this issue

File: composer.json (hunk 2 above)
Comment (oshmyheliuk) [reply composer.json:29]: added

File: .travis.yml (hunk 3 above)
Comment (oshmyheliuk) [reply .travis.yml]: updated

**Review by NadiyaS — APPROVED** [review]

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  - if [ $TRAVIS_SECURE_ENV_VARS == "true" ]; then composer config http-basic.repo.magento.com ${REPO_USERNAME} ${REPO_PASSWORD}; fi;
+  - if [ $TRAVIS_SECURE_ENV_VARS == "true" ]; then composer config github-oauth.github.com ${GITHUB_TOKEN}; fi;
+  - composer config repositories.magento composer https://repo.magento.com/
+  - composer require "magento/framework:*" --no-update
+  - composer require "magento/module-store:*" --no-update
+  - composer require "magento/module-url-rewrite:*" --no-update
+  - composer update -n --no-suggest
+
+script:
+  - if [ "$TEST_SUITE" == "static-unit" ]; then ./Test/static/static-travis.sh; fi
```
Comment (shiftedreality) [inline .travis.yml:25]: This must be in `tests` directory

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 5 above)
Comment (oshmyheliuk) [reply .travis.yml:25]: Why `must`? it will be strange to have `Test` and `tests` directories on one level.

**Review by shiftedreality — APPROVED** [review]

**Pull request merged**